tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -443,3 +443,28 @@ def _discover(obj):
     if expected_top != count:
         raise TestFailure("Expected %d found %d for cosLut" % (expected_top, count))
 
+@cocotb.test(skip=cocotb.LANGUAGE in ["vhdl"])
+def type_check_verilog(dut):
+    """
+    Test if types are recognized
+    """
+
+    tlog = logging.getLogger("cocotb.test")
+
+    yield Timer(10)
+
+    test_handles = [
+        (dut.stream_in_ready, "GPI_REGISTER"),
+        (dut.register_array, "GPI_ARRAY"),
+        (dut.NUM_OF_MODULES, "GPI_PARAMETER"),
+        (dut.temp, "GPI_REGISTER"),
+        (dut.and_output, "GPI_NET"),
+        (dut.stream_in_data, "GPI_NET"),
+    ]
```

Comment:
Can we also have a  test with `logic`?

Reply:
I have added. 

`logic` should default to `vpiNet` but for vcs it is becoming `GPI_MODULE` so the test fails (all others seem happy).

Reply:
You modified an input port to explicitly specify `logic` as net type. In SV, the default is already `logic` unless overwritten with the default_nettype compiler directive. 

The story is different for variables/nets within the design. Can you add a test for such an element as well (logic, wire and reg)?

If VCS behaves differently, can you exclude the test from VCS (see test_cocotb.py for examples on how to do that). Ideally you can note the version number of VCS that fails to allow us to revisit this exception in the future more easily.

Reply:
If I do this:
```systemverilog
logic logic_a, logic_b, logic_c;
assign logic_a = stream_in_valid;
always@* logic_b = stream_in_valid;
always@(posedge clk) logic_c <= stream_in_valid;
```
In icarus `logic_a` is `vpiNet` and `logic_b`/`logic_c` `vpiReg` in all other simulators it will be all `vpiReg` .

Any opinion on how to handle this?

Reply:
Please special-case icarus in the code, but test for the existing behavior for both icarus and other simulators. And file an icarus upstream bug and reference the bug number here.